### PR TITLE
disable sync committee

### PR DESCRIPTION
**Pull request by sb-anderson**
### Disable sync committee
- Spec : [546669](https://github.com/superblock-dev/chronos-spec/commit/5466699bc422910583b93d532e774b2c1c470436)
- Remove sync sign aggregation in block
- Remove sync committee feature in state transition
- Remove sync subnet subscription
- Remove sync committee apis

### CODE REVIEW

File: beacon-chain/core/helpers/private_access_fuzz_noop_test.go [hunk 1]
```diff
@@ -4,14 +4,6 @@ package helpers
 
 import "github.com/prysmaticlabs/prysm/v5/beacon-chain/cache"
 
-func CommitteeCache() *cache.FakeCommitteeCache {
```
Comment (syjn99) [inline beacon-chain/core/helpers/private_access_fuzz_noop_test.go]: It seems this deletion is not appropriate for current scope of work

**Comment by syjn99** [comment]
General review:

- Commented lines should be removed. (e.g. `lightClientEndpoints`)
- `sync_committee` and `sync committee` should be removed (e.g.  commented one, `ssz_proto_library.bzl`)
- `SyncAggregate` should be removed. (search for `syncaggregate`)
- `LightClient` should be removed. (search for regex `light*client`)
- `ContributionAndProof` should be removed. (search for `contributionandproof`)
- Running Gazelle
- `syncnets` in `MetaDataV1` (`p2p_messages.proto`)
- `Eth2FastAggregateVerify`(`eth2_fast_aggregate_verify`): used in `process_sync_aggregate`

File: proto/prysm/v1alpha1/health.pb.go [hunk 2]
```diff
@@ -8,16 +8,15 @@ package eth
...
-	sync "sync"
-
 	_ "google.golang.org/genproto/googleapis/api/annotations"
 	grpc "google.golang.org/grpc"
 	codes "google.golang.org/grpc/codes"
 	status "google.golang.org/grpc/status"
 	protoreflect "google.golang.org/protobuf/reflect/protoreflect"
 	protoimpl "google.golang.org/protobuf/runtime/protoimpl"
 	emptypb "google.golang.org/protobuf/types/known/emptypb"
+	reflect "reflect"
```
Comment (syjn99) [inline proto/prysm/v1alpha1/health.pb.go]: "`goimports` command not found" significantly increases the number of file changes. Should rerun pb.go file generator without that error

File: consensus-types/interfaces/light_client.go [hunk 3]
```diff
@@ -1,61 +0,0 @@
...
-	ssz "github.com/prysmaticlabs/fastssz"
-	fieldparams "github.com/prysmaticlabs/prysm/v5/config/fieldparams"
-	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
-	pb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
-)
-
-type LightClientExecutionBranch = [fieldparams.ExecutionBranchDepth][fieldparams.RootLength]byte
-type LightClientSyncCommitteeBranch = [fieldparams.SyncCommitteeBranchDepth][fieldparams.RootLength]byte
-type LightClientSyncCommitteeBranchElectra = [fieldparams.SyncCommitteeBranchDepthElectra][fieldparams.RootLength]byte
-type LightClientFinalityBranch = [fieldparams.FinalityBranchDepth][fieldparams.RootLength]byte
```
Comment (syjn99) [inline consensus-types/interfaces/light_client.go:13]: As those were deleted, `ExecutionBranchDepth` and `FinalityBranchDepth` are no longer used

File: consensus-types/blocks/kzg.go [hunk 4]
```diff
@@ -12,7 +12,7 @@ import (
 )
 
 const (
-	bodyLength    = 13 // The number of elements in the BeaconBlockBody Container for Electra
+	bodyLength    = 12 // The number of elements in the BeaconBlockBody Container for Electra
 	logBodyLength = 4  // The log 2 of bodyLength
 	kzgPosition   = 11 // The index of the KZG commitment list in the Body
 	kzgRootIndex  = 54 // The Merkle index of the KZG commitment list's root in the Body's Merkle tree
```
Comment (syjn99) [inline consensus-types/blocks/kzg.go]: Those values must be adjusted as a field of beacon block body is deleted.

kzgPosition = 10
kzgRootIndex = 52

This test will be passed if then
```
=== RUN   TestInclusionProofs
    assertions.go:144: deneb_test.go:14 Unexpected error: invalid KZG commitment inclusion proof
```

File: consensus-types/blocks/kzg.go [hunk 5]
```diff
@@ -195,17 +195,6 @@ func topLevelRoots(body interfaces.ReadOnlyBeaconBlockBody) ([][]byte, error) {
 	}
 	copy(layer[7], root[:])
 
-	// Sync Aggregate
```
Comment (syjn99) [inline consensus-types/blocks/kzg.go:198]: copying each field after sync aggregate must be fixed

File: beacon-chain/p2p/discovery_test.go [hunk 6]
```diff
@@ -596,7 +591,7 @@ func TestRefreshENR_ForkBoundaries(t *testing.T) {
 			},
 			postValidation: func(t *testing.T, s *Service) {
 				assert.Equal(t, version.Altair, s.metaData.Version())
-				assert.DeepEqual(t, bitfield.Bitvector4{0x00}, s.metaData.MetadataObjV1().Syncnets)
+				// assert.DeepEqual(t, bitfield.Bitvector4{0x00}, s.metaData.MetadataObjV1().Syncnets)
```
Comment (gts2030) [inline beacon-chain/p2p/discovery_test.go]: This commented line can be removed.

File: beacon-chain/p2p/discovery_test.go [hunk 7]
```diff
@@ -629,7 +624,7 @@ func TestRefreshENR_ForkBoundaries(t *testing.T) {
...
```
Comment (gts2030) [inline beacon-chain/p2p/discovery_test.go]: This also can be removed.

File: beacon-chain/p2p/discovery_test.go [hunk 8]
```diff
@@ -667,12 +662,11 @@ func TestRefreshENR_ForkBoundaries(t *testing.T) {
 				s.metaData = wrapper.WrappedMetadataV0(new(ethpb.MetaDataV0))
 				s.updateSubnetRecordWithMetadata([]byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00})
 				cache.SubnetIDs.AddPersistentCommittee([]uint64{1, 2, 3, 23}, 0)
-				cache.SyncSubnetIDs.AddSyncCommitteeSubnets([]byte{'A'}, 0, []uint64{0, 1}, 0)
 				return s
 			},
 			postValidation: func(t *testing.T, s *Service) {
 				assert.Equal(t, version.Altair, s.metaData.Version())
-				assert.DeepEqual(t, bitfield.Bitvector4{0x03}, s.metaData.MetadataObjV1().Syncnets)
+				// assert.DeepEqual(t, bitfield.Bitvector4{0x03}, s.metaData.MetadataObjV1().Syncnets)
```
Comment (gts2030) [inline beacon-chain/p2p/discovery_test.go]: This can be removed too.

**Comment by gts2030** [comment]
In beacon-chain/p2p/gossip_scoring_params.go, I think [this part](https://github.com/overprotocol/chronos/blob/d81a36ccb63443e2a5b071bd50c935e0d63ebcf5/beacon-chain/p2p/gossip_scoring_params.go#L28) can be removed.
```
// syncContributionWeight specifies the scoring weight that we apply to
// our sync contribution topic.
syncContributionWeight = 0.2
```
And I have a question that after removing those weights, does other weights are okey to use without any change?

File: beacon-chain/p2p/gossip_scoring_params.go [hunk 9]
```diff
@@ -560,7 +448,7 @@ func scoreByWeight(weight, threshold float64) float64 {
 // maxScore attainable by a peer.
 func maxScore() float64 {
 	totalWeight := beaconBlockWeight + aggregateWeight + syncContributionWeight +
-		attestationTotalWeight + syncCommitteesTotalWeight + attesterSlashingWeight +
```
Comment (gts2030) [inline beacon-chain/p2p/gossip_scoring_params.go:563]: syncContributionWeight seems to be removed too.

File: beacon-chain/p2p/gossip_topic_mappings_test.go [hunk 10]
```diff
@@ -53,9 +53,6 @@ func TestGossipTopicMappings_CorrectType(t *testing.T) {
 	pMessage = GossipTopicMappings(AttesterSlashingSubnetTopicFormat, 0)
 	_, ok = pMessage.(*ethpb.AttesterSlashing)
 	assert.Equal(t, true, ok)
-	pMessage = GossipTopicMappings(AggregateAndProofSubnetTopicFormat, 0)
-	_, ok = pMessage.(*ethpb.SignedAggregateAttestationAndProof)
-	assert.Equal(t, true, ok)
```
Comment (gts2030) [inline beacon-chain/p2p/gossip_topic_mappings_test.go]: Why does this test case is removed here?
And is it okey to leave `AggregateAndProofSubnetTopicFormat` test cases in other forks?

File: beacon-chain/rpc/core/BUILD.bazel [hunk 11]
```diff
@@ -48,16 +47,16 @@ go_library(
...
-    deps = [
-        "//beacon-chain/cache:go_default_library",
-        "//config/params:go_default_library",
-        "//consensus-types/validator:go_default_library",
-        "//proto/prysm/v1alpha1:go_default_library",
-        "//testing/assert:go_default_library",
-        "//testing/require:go_default_library",
-    ],
-)
+# go_test(
```
Comment (gts2030) [inline beacon-chain/rpc/core/BUILD.bazel]: Commented lines are left here.

**Comment by sb-anderson** [comment]
> In beacon-chain/p2p/gossip_scoring_params.go, I think [this part](https://github.com/overprotocol/chronos/blob/d81a36ccb63443e2a5b071bd50c935e0d63ebcf5/beacon-chain/p2p/gossip_scoring_params.go#L28) can be removed.
> 
> ```
> // syncContributionWeight specifies the scoring weight that we apply to
> // our sync contribution topic.
> syncContributionWeight = 0.2
> ```
> 
> And I have a question that after removing those weights, does other weights are okey to use without any change?

It seems fine because the weight value is a relative value that hasn’t been normalized.

File: beacon-chain/rpc/prysm/v1alpha1/validator/attester_test.go [hunk 12]
```diff
@@ -543,102 +542,102 @@ func TestGetAttestationData_CommitteeIndexIsZeroPostElectra(t *testing.T) {
...
-	for i := primitives.Slot(100); i < 200; i++ {
-		subnets := cache.SubnetIDs.GetAttesterSubnetIDs(i)
-		assert.Equal(t, 1, len(subnets))
-		if isAggregator[i-100] {
-			subnets = cache.SubnetIDs.GetAggregatorSubnetIDs(i)
-			assert.Equal(t, 1, len(subnets))
-		}
-	}
-}
+// func TestServer_SubscribeCommitteeSubnets_NoSlots(t *testing.T) {
```
Comment (gts2030) [inline beacon-chain/rpc/prysm/v1alpha1/validator/attester_test.go]: Isn't this test is about attestation subnet?

File: beacon-chain/rpc/prysm/v1alpha1/validator/blocks_test.go [hunk 13]
```diff
@@ -70,176 +67,6 @@ func TestServer_StreamAltairBlocks_ContextCanceled(t *testing.T) {
 	exitRoutine <- true
 }
 
-func TestServer_StreamAltairBlocks_OnHeadUpdated(t *testing.T) {
```
Comment (gts2030) [inline beacon-chain/rpc/prysm/v1alpha1/validator/blocks_test.go]: Does this test should be removed? Just removing SetCurrentSyncCommittee seemes to be okey.

File: testing/endtoend/evaluators/validator.go [hunk 14]
```diff
@@ -21,16 +21,14 @@ import (
...
 	"google.golang.org/grpc"
-	"google.golang.org/protobuf/types/known/emptypb"
 )
 
 var expectedParticipation = 0.99
 
 var expectedMulticlientParticipation = 0.95
 
-var expectedSyncParticipation = 0.99
+// var expectedSyncParticipation = 0.99
```
Comment (gts2030) [inline testing/endtoend/evaluators/validator.go]: Commented line left here

File: validator/client/beacon-api/duties_test.go [hunk 15]
```diff
@@ -1029,17 +835,17 @@ func TestGetDuties_Valid(t *testing.T) {
...
-				dutiesProvider.EXPECT().SyncDuties(
-					ctx,
-					testCase.epoch,
-					validatorIndices,
-				).Return(
-					generateValidSyncDuties(pubkeys, validatorIndices),
-					nil,
-				).Times(2)
-			}
+			// fetchSyncDuties := testCase.epoch >= params.BeaconConfig().AltairForkEpoch
```
Comment (gts2030) [inline validator/client/beacon-api/duties_test.go]: Commented line left here.

**Comment by gts2030** [comment]
- Unremoved part
[beacon-chain/state/stateutil/BUILD.bazel:18](https://github.com/overprotocol/chronos/blob/00a7e0235b8b73f9f663af034cf4a348154df19e/beacon-chain/state/stateutil/BUILD.bazel#L18) -> sync_committee.root.go
[beacon-chain/p2p/pubsub_filter.go:20](https://github.com/overprotocol/chronos/blob/00a7e0235b8b73f9f663af034cf4a348154df19e/beacon-chain/p2p/pubsub_filter.go#L20) -> syncContribution in comment
[beacon-chain/rpc/endpoints_test.go:96](https://github.com/overprotocol/chronos/blob/00a7e0235b8b73f9f663af034cf4a348154df19e/beacon-chain/rpc/endpoints_test.go#L96) -> contribution_and_proofs
[beacon-chain/rpc/eth/config/handlers_test.go:376](https://github.com/overprotocol/chronos/blob/00a7e0235b8b73f9f663af034cf4a348154df19e/beacon-chain/rpc/eth/config/handlers_test.go#L376) -> DOMAIN_CONTRIBUTION_AND_PROOF
[beacon-chain/rpc/eth/config/handlers_test.go:416](https://github.com/overprotocol/chronos/blob/00a7e0235b8b73f9f663af034cf4a348154df19e/beacon-chain/rpc/eth/config/handlers_test.go#L416) -> MAX_REQUEST_LIGHT_CLIENT_UPDATES

- Should be changed
[beacon-chain/p2p/subnets_test.go:315, 329 ](https://github.com/overprotocol/chronos/blob/00a7e0235b8b73f9f663af034cf4a348154df19e/beacon-chain/p2p/subnets_test.go#L315)-> syncSubnets to attSubnets

File: beacon-chain/rpc/prysm/v1alpha1/beacon/validators_test.go [hunk 16]
```diff
@@ -1675,66 +1673,6 @@ func TestServer_GetValidatorParticipation_OrphanedUntilGenesis(t *testing.T) {
 	assert.DeepEqual(t, wanted, res.Participation, "Incorrect validator participation respond")
 }
 
-func TestServer_GetValidatorParticipation_CurrentAndPrevEpochWithBits(t *testing.T) {
```
Comment (syjn99) [inline beacon-chain/rpc/prysm/v1alpha1/beacon/validators_test.go]: 1. This test looks fine except sync committee part. Is there any reason to remove the whole test?
2. If we decided to remove this test, `runGetValidatorParticipationCurrentAndPrevEpoch` also can be removed.

File: beacon-chain/state/stateutil/sync_committee.root.go [hunk 17]
```diff
@@ -4,42 +4,8 @@ import (
...
-	aggregateKeyRoot, err := merkleizePubkey(committee.AggregatePubkey)
-	if err != nil {
-		return [32]byte{}, err
-	}
-	fieldRoots = [][32]byte{pubkeyRoot, aggregateKeyRoot}
-
-	return ssz.BitwiseMerkleize(fieldRoots, uint64(len(fieldRoots)), uint64(len(fieldRoots)))
-}
-
 func merkleizePubkey(pubkey []byte) ([32]byte, error) {
```
Comment (syjn99) [inline beacon-chain/state/stateutil/sync_committee.root.go]: This function can be moved to `validator_root.go`, and this file can be deleted.

File: beacon-chain/db/kv/schema.go [hunk 18]
```diff
@@ -18,7 +18,6 @@ var (
 	registrationBucket    = []byte("registration")
 
 	// Light Client Updates Bucket
```
Comment (syjn99) [inline beacon-chain/db/kv/schema.go]: This comment can be removed.

File: config/params/mainnet_config.go [hunk 19]
```diff
@@ -184,11 +180,11 @@ var mainnetBeaconConfig = &BeaconChainConfig{
...
 	BeaconStateFieldCount:          22,
-	BeaconStateAltairFieldCount:    25,
-	BeaconStateBellatrixFieldCount: 26,
-	BeaconStateCapellaFieldCount:   29,
-	BeaconStateDenebFieldCount:     29,
-	BeaconStateElectraFieldCount:   35,
+	BeaconStateAltairFieldCount:    23,
+	BeaconStateBellatrixFieldCount: 24,
+	BeaconStateCapellaFieldCount:   27,
+	BeaconStateDenebFieldCount:     33,
```
Comment (syjn99) [inline config/params/mainnet_config.go]: Deneb field count should be same with capella one

File: consensus-types/blocks/proofs_test.go [hunk 20]
```diff
@@ -41,7 +41,7 @@ func TestComputeBlockBodyFieldRoots_Altair(t *testing.T) {
 
 	fieldRoots, err := ComputeBlockBodyFieldRoots(context.Background(), b)
 	require.NoError(t, err)
-	trie, err := trie.GenerateTrieFromItems(fieldRoots, 4)
+	trie, err := trie.GenerateTrieFromItems(fieldRoots, 3)
```
Comment (syjn99) [inline consensus-types/blocks/proofs_test.go:44]: Question: what's the reason for this change?

**Review by syjn99 — COMMENTED** [review]
Seems some `BUILD.bazel` files are not updated. This command will help:

```bash
bazel run //:gazelle .
```

File: proto/prysm/v1alpha1/validator.proto [hunk 21]
```diff
@@ -830,13 +750,6 @@ message AssignValidatorToSubnetRequest {
...
 }
 
-message AggregatedSigAndAggregationBitsRequest {
-    repeated SyncCommitteeMessage msgs = 1;
-    uint64 slot = 2 [(ethereum.eth.ext.cast_type) = "github.com/prysmaticlabs/prysm/v5/consensus-types/primitives.Slot"];
-    uint64 subnet_id = 3;
-    bytes block_root = 4 [(ethereum.eth.ext.ssz_size) = "32"];
-}
-
 message AggregatedSigAndAggregationBitsResponse {
```
Comment (syjn99) [inline proto/prysm/v1alpha1/validator.proto]: This can be also removed

File: beacon-chain/p2p/utils.go [hunk 22]
```diff
@@ -180,21 +179,22 @@ func metaDataFromFile(path string) (metadata.Metadata, error) {
...
+	// 	// If unmarshal failed, try to unmarshal for V0
+	// 	log.WithError(err).Info("Error unmarshalling V1 metadata from file, try to unmarshal for V0.")
+	// 	md0 := &pb.MetaDataV0{}
+	// 	md0Err := md0.UnmarshalSSZ(src)
+	// 	if md0Err != nil {
+	// 		log.WithError(md0Err).Error("Error unmarshalling V0 metadata from file")
+	// 		return nil, md0Err
+	// 	}
+	// 	return wrapper.WrappedMetadataV0(md0), nil
+	// }
```
Comment (syjn99) [inline beacon-chain/p2p/utils.go]: These codes can be removed

File: beacon-chain/p2p/discovery.go [hunk 23]
```diff
@@ -158,36 +156,12 @@ func (s *Service) RefreshENR() {
...
-			log.WithError(err).Error("Could not retrieve sync bitfield")
-			return
-		}
-		if bytes.Equal(bitV, currentBitV) && bytes.Equal(bitS, currentBitS) &&
-			s.Metadata().Version() == version.Altair {
-			// return early if bitfields haven't changed
-			return
-		}
-		s.updateSubnetRecordWithMetadataV2(bitV, bitS)
+	// Phase 0 behaviour.
```
Comment (syjn99) [inline beacon-chain/p2p/discovery.go]: Sorry for splitting hairs, as we use metadata without sync bits, this comment can be confused

File: beacon-chain/p2p/gossip_topic_mappings.go [hunk 24]
```diff
@@ -55,11 +53,6 @@ func GossipTopicMappings(topic string, epoch primitives.Epoch) proto.Message {
 			return &ethpb.AttesterSlashingElectra{}
 		}
 		return gossipMessage(topic)
-	case AggregateAndProofSubnetTopicFormat:
```
Comment (syjn99) [inline beacon-chain/p2p/gossip_topic_mappings.go]: I wonder why this case is removed

File: encoding/ssz/detect/configfork_test.go [hunk 25]
```diff
@@ -297,7 +297,7 @@ func TestDetectAndUnmarshalBlock(t *testing.T) {
 			name:      "genesis block in altair slot",
 			b:         signedTestBlockGenesis,
 			slot:      bellaS - 1,
-			errExists: true,
+			errExists: false,
```
Comment (syjn99) [inline encoding/ssz/detect/configfork_test.go:300]: What's the reason for this change?

File: beacon-chain/rpc/endpoints_test.go [hunk 26]
```diff
@@ -102,10 +92,7 @@ func Test_endpoints(t *testing.T) {
 		"/eth/v1/validator/aggregate_attestation":          {http.MethodGet},
 		"/eth/v1/validator/aggregate_and_proofs":           {http.MethodPost},
 		"/eth/v1/validator/beacon_committee_subscriptions": {http.MethodPost},
-		"/eth/v1/validator/sync_committee_subscriptions":   {http.MethodPost},
 		"/eth/v1/validator/beacon_committee_selections":    {http.MethodPost},
-		"/eth/v1/validator/sync_committee_selections":      {http.MethodPost},
-		"/eth/v1/validator/sync_committee_contribution":    {http.MethodGet},
 		"/eth/v1/validator/contribution_and_proofs":        {http.MethodPost},
```
Comment (syjn99) [inline beacon-chain/rpc/endpoints_test.go]: This can be also removed

File: beacon-chain/p2p/pubsub_filter.go [hunk 27]
```diff
@@ -17,7 +17,6 @@ var _ pubsub.SubscriptionFilter = (*Service)(nil)
 // It is set at this limit to handle the possibility
 // of double topic subscriptions at fork boundaries.
 // -> 64 Attestation Subnets * 2.
-// -> 4 Sync Committee Subnets * 2.
 // -> Block,Aggregate,ProposerSlashing,AttesterSlashing,Exits,SyncContribution * 2.
```
Comment (syjn99) [inline beacon-chain/p2p/pubsub_filter.go]: SyncContribution can also be removed. + I wonder how the limit(`200`) is calculated for the first time, but it seems fine to remain it unchanged.

File: consensus-types/blocks/proofs_test.go (hunk 20 above)
Comment (sb-anderson) [reply consensus-types/blocks/proofs_test.go:44]: As sync committee related fields are deleted, len(fieldRoots) is also decreased (9 -> 8). so trie depth is decreased(4 -> 3).

File: encoding/ssz/detect/configfork_test.go (hunk 25 above)
Comment (sb-anderson) [reply encoding/ssz/detect/configfork_test.go:300]: As the block structrure of altair is same as phase 0, genesis block can be processed in altair slot

File: proto/eth/v2/beacon_block.proto
Comment (syjn99) [inline proto/eth/v2/beacon_block.proto:1]: I think each field number for `BeaconBlockBody` should be same with `prysm/v1alpha1/beacon_block.proto`. Field numbers are rearranged at `prysm/v1alpha1/beacon_block.proto`, but this file has no change.

File: beacon-chain/sync/service.go [hunk 28]
```diff
@@ -73,7 +69,7 @@ var (
 )
 
 // Common type for functional p2p validation options.
-type validationFn func(ctx context.Context) (pubsub.ValidationResult, error)
+// type validationFn func(ctx context.Context) (pubsub.ValidationResult, error)
```
Comment (syjn99) [inline beacon-chain/sync/service.go]: Comment can be removed

File: beacon-chain/rpc/eth/beacon/handlers_test.go [hunk 29]
```diff
@@ -1232,11 +1232,14 @@ func TestPublishBlock(t *testing.T) {
 	t.Run("Phase 0", func(t *testing.T) {
```
Comment (syjn99) [inline beacon-chain/rpc/eth/beacon/handlers_test.go]: This test is broken with panicking:

```
--- FAIL: TestPublishBlockSSZ (0.00s)
    --- FAIL: TestPublishBlockSSZ/Phase_0 (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x103f2c1d4]
```

File: beacon-chain/sync/decode_pubsub.go [hunk 30]
```diff
@@ -41,8 +41,6 @@ func (s *Service) decodePubsubMessage(msg *pubsub.Message) (ssz.Unmarshaler, err
 		topic = p2p.GossipTypeMapping[reflect.TypeOf(&ethpb.Attestation{})]
 		// Given that both sync message related subnets have the same message name, we have to
```
Comment (syjn99) [inline beacon-chain/sync/decode_pubsub.go]: This comment can be removed

File: beacon-chain/sync/rpc_metadata.go [hunk 31]
```diff
@@ -113,16 +93,7 @@ func (s *Service) sendMetaDataRequest(ctx context.Context, id peer.ID) (metadata
 		return nil, err
 	}
 	// Defensive check to ensure valid objects are being sent.
```
Comment (syjn99) [inline beacon-chain/sync/rpc_metadata.go]: This comment can be removed

File: beacon-chain/rpc/eth/beacon/handlers_test.go (hunk 29 above)
Comment (sb-anderson) [reply beacon-chain/rpc/eth/beacon/handlers_test.go]: Fixed

File: validator/client/beacon-api/duties_test.go (hunk 15 above)
Comment (syjn99) [reply validator/client/beacon-api/duties_test.go]: @sb-anderson 
This comment is not resolved yet

**Review by syjn99 — APPROVED** [review]
LGTM!
